### PR TITLE
Travis CI support

### PR DESCRIPTION
**Pull request by lebdron**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+install:
+  - DEPS_DIR="${TRAVIS_BUILD_DIR}/deps"
+  - mkdir -p "${DEPS_DIR}" && cd "${DEPS_DIR}"
+  ############################################################################
+  # Install a recent CMake (unless already installed on OS X)
+  ############################################################################
+  - |
+    if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+      if [[ -z "$(ls -A ${DEPS_DIR}/cmake/bin)" ]]; then
+        CMAKE_URL="https://cmake.org/files/v3.8/cmake-3.8.0-rc2-Linux-x86_64.tar.gz"
```
Comment (l4l) [inline .travis.yml]: Why 3.8? Project requires 3.3.0

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      export PATH="${DEPS_DIR}/cmake/bin:${PATH}"
+    else
+      if ! brew ls --version cmake &>/dev/null; then brew install cmake; fi
+    fi
+
+script:
+  - cd "${TRAVIS_BUILD_DIR}"
+  - mkdir build; cd build
+  - export CC=$COMPILERCC; export CXX=$COMPILERCXX
+  - cmake .. && make && ctest
```
Comment (l4l) [inline .travis.yml]: instead write:
`cmake .. && make && CTEST_OUTPUT_ON_FAILURE=1 ctest`
so we'll able to see tests fails logs

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  include:
+    - os: linux
+      compiler: gcc
+      addons:
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['gcc-5', 'g++-5', 'xsltproc', 'libgtest-dev']
+      env:
+        - COMPILERCC=gcc-5
+        - COMPILERCXX=g++-5
```
Comment (l4l) [inline .travis.yml]: why not `CC`/`CXX` directly?

File: .travis.yml (hunk 1 above)
Comment (lebdron) [inline .travis.yml]: https://github.com/hyperledger/iroha-ametsuchi/blob/master/circle.yml#L29

Comment (l4l) [inline .travis.yml]: ah, ok
But anyways that should work w/ 3.3

**Pull request merged**